### PR TITLE
Add placeholder variables for IPH physical heat sink model

### PR DESCRIPTION
**Pull request by taylorbrown75**
Added placeholder variables to tower, trough, and mslf IPH cmods for use by future physical heat sink model.

### CODE REVIEW

File: ssc/cmod_fresnel_physical_iph.cpp [hunk 1]
```diff
@@ -153,7 +153,17 @@ static var_info _cm_vtab_fresnel_physical_iph[] = {
 
     // Heat Sink
 
-    {SSC_INPUT,     SSC_NUMBER,         "pb_pump_coef",                "Pumping power to move 1kg of HTF through PB loop",                                      "kW/kg",               "",                              "Heat Sink",           "*",                "",                 ""},
+    { SSC_INPUT,    SSC_NUMBER,         "pb_pump_coef",                "Pumping power to move 1kg of HTF through PB loop",                                      "kW/kg",               "",                             "Heat Sink",            "*",                "",                 "" },
```
Comment (sjanzou) [inline ssc/cmod_fresnel_physical_iph.cpp:156]: This value is pre-existing and has a value in all default configurations - correct?

File: ssc/cmod_fresnel_physical_iph.cpp [hunk 2]
```diff
@@ -153,7 +153,17 @@ static var_info _cm_vtab_fresnel_physical_iph[] = {
...
+
+    { SSC_INPUT,    SSC_NUMBER,         "hs_type",                     "0: ideal model, 1: physical steam model",                                               "",                    "",                             "Heat Sink",            "?=0",              "",                 "" },
```
Comment (sjanzou) [inline ssc/cmod_fresnel_physical_iph.cpp:158]: This value defaults to zero and cannot be set in the UI currently - correct?

File: ssc/cmod_fresnel_physical_iph.cpp [hunk 3]
```diff
@@ -940,6 +955,12 @@ class cm_fresnel_physical_iph : public compute_module
             c_heat_sink.mc_reported_outputs.assign(C_pc_heat_sink::E_M_DOT_HTF, allocate("m_dot_htf_heat_sink", n_steps_fixed), n_steps_fixed);
             c_heat_sink.mc_reported_outputs.assign(C_pc_heat_sink::E_T_HTF_IN, allocate("T_heat_sink_in", n_steps_fixed), n_steps_fixed);
             c_heat_sink.mc_reported_outputs.assign(C_pc_heat_sink::E_T_HTF_OUT, allocate("T_heat_sink_out", n_steps_fixed), n_steps_fixed);
+
+            c_heat_sink_pointer = &c_heat_sink;
+        }
+        else
+        {
+            throw exec_error("fresnel_physical_iph", "hs_type != 0; other heat sink models are not currently supported");
```
Comment (sjanzou) [inline ssc/cmod_fresnel_physical_iph.cpp:963]: This catches any non-implemented error in the release - correct?

File: ssc/cmod_fresnel_physical_iph.cpp [hunk 4]
```diff
@@ -1087,7 +1108,7 @@ class cm_fresnel_physical_iph : public compute_module
         // Instantiate Solver
         C_csp_solver csp_solver(weather_reader,
             c_fresnel,
-            c_heat_sink,
+            *c_heat_sink_pointer,
```
Comment (sjanzou) [inline ssc/cmod_fresnel_physical_iph.cpp:1111]: Is this checked for null now that a pointer is used?

**Review by sjanzou — APPROVED** [review]
Looks and tests okay and seems to have all values for upgrading in a patch.

File: ssc/cmod_fresnel_physical_iph.cpp (hunk 2 above)
Comment (taylorbrown75) [reply ssc/cmod_fresnel_physical_iph.cpp:158]: Correct, hs_type defaults to zero and cannot be set via UA

File: ssc/cmod_fresnel_physical_iph.cpp (hunk 3 above)
Comment (taylorbrown75) [reply ssc/cmod_fresnel_physical_iph.cpp:963]: That is correct

File: ssc/cmod_fresnel_physical_iph.cpp (hunk 1 above)
Comment (taylorbrown75) [reply ssc/cmod_fresnel_physical_iph.cpp:156]: That is correct, no changes were made to 'pb_pump_coef'

File: ssc/cmod_fresnel_physical_iph.cpp (hunk 4 above)
Comment (taylorbrown75) [reply ssc/cmod_fresnel_physical_iph.cpp:1116]: Added nullptr check ([850d489](https://github.com/NREL/ssc/pull/1250/commits/850d48936bb946817e914ceddf7b38d4df072f16))

**Pull request merged**